docs/modules/third_party_rules_callbacks.md

```diff
@@ -148,6 +148,44 @@ deny an incoming event, see [`check_event_for_spam`](spam_checker_callbacks.md#c
 
 If multiple modules implement this callback, Synapse runs them all in order.
 
+### `on_profile_update`
```

Comment:
Is it possible to distinguish in this method whether a user was deactivated versus simply setting their profile information to a display name of `""` and an avatar URL of `None`? I could see implementations of this method commonly having a conditional of:

```python
if new_profile.display_name == "" and new_profile.avatar_url is None:
    return
```

in which case users would be able to trigger a profile update without this code catching it.

Can we track whether a profile update is the result of a deactivation and refrain from running this method in that instance?

Reply:
> Can we track whether a profile update is the result of a deactivation and refrain from running this method in that instance?

I don't think we should skip this method in this case. It sounds to me like a way to get modules out of sync with a user's profile (e.g. if they maintain their own db table for profiles, and the user gets reactivated and changes just their display name, the module will still think they've got their old avatar).
We could pass a flag onto the profile handler then to the callback. The initial intention was to have modules react to deactivation separately in `on_deactivation`, but I guess it doesn't hurt to have this information available there as well.

Reply:
(done in https://github.com/matrix-org/synapse/pull/12062/commits/698afcf29a50f644423d587bbb04923249ecd9db)